pkg/cmdutil/repo_factory.go

```diff
@@ -0,0 +1,73 @@
+package cmdutil
```

Comment:
I named this `repo_factory` as I was envisioning a future when the all the constructors in `ghrepo` can be moved here and this file can be responsible for all repo creating.